libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3272,48 +3272,33 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 				if (rational->isFractional())
 					solAssert(toCategory == Type::Category::FixedPoint, "");
 
-			if (toCategory == Type::Category::FixedBytes)
-			{
-				FixedBytesType const& toBytesType = dynamic_cast<FixedBytesType const&>(_to);
-				body =
-					Whiskers("converted := <shiftLeft>(<clean>(value))")
-					("shiftLeft", shiftLeftFunction(256 - toBytesType.numBytes() * 8))
-					("clean", cleanupFunction(_from))
-					.render();
-			}
-			else if (toCategory == Type::Category::Enum)
-				body =
-					Whiskers("converted := <cleanEnum>(<cleanInt>(value))")
-					("cleanEnum", cleanupFunction(_to))
-					("cleanInt", cleanupFunction(_from))
-					.render();
-			else if (toCategory == Type::Category::FixedPoint)
-				solUnimplemented("Not yet implemented - FixedPointType.");
-			else if (toCategory == Type::Category::Address || toCategory == Type::Category::Contract)
+			if (toCategory == Type::Category::Address || toCategory == Type::Category::Contract)
 				body =
 					Whiskers("converted := <convert>(value)")
 					("convert", conversionFunction(_from, IntegerType(160)))
 					.render();
-			else if (toCategory == Type::Category::Integer)
+			else
 			{
-				IntegerType const& to = dynamic_cast<IntegerType const&>(_to);
-
-				// Clean according to the "to" type, except if this is
-				// a widening conversion.
-				IntegerType const* cleanupType = &to;
-				if (fromCategory == Type::Category::Integer)
+				Whiskers bodyTemplate(R"(
+					value := <cleanInput>(value)
+					converted := <cleanOutput>(<convert>)
```

Comment:
I think this would be easier to read if `value` was in the template. `<convert>` could be an empty string when not used. Now you have to read this whole `if` branch and then backtrack to read this again.
```suggestion
					converted := <cleanOutput>(<convert>(value))
```
`(value)` will be optimized into `value` anyway, right?

Reply:
No, `(value)` is an invalid expression, but you are right, it would be better to have `convert` always be exactly one function.